integration_tests/blockstack_integration_tests/scenarios/name_pre_reg_v2_upgrade_migration.py

```diff
@@ -0,0 +1,110 @@
+    testlib.Wallet( "5Kg5kJbQHvk1B64rJniEmgbD83FpZpbw2RjdAZEzTefs9ihN3Bz", 100000000000 ),
+    testlib.Wallet( "5JuVsoS9NauksSkqEjbUZxWwgGDQbMwPsEfoRBSpLpgDX1RtLX7", 100000000000 ),
+    testlib.Wallet( "5KEpiSRr1BrT8vRD7LKGCEmudokTh1iMHbiThMQpLdwBwhDJB1T", 100000000000 ),
+    testlib.Wallet( "5K5hDuynZ6EQrZ4efrchCwy6DLhdsEzuJtTDAf3hqdsCKbxfoeD", 100000000000 ),
+    testlib.Wallet( "5J39aXEeHh9LwfQ4Gy5Vieo7sbqiUMBXkPH7SaMHixJhSSBpAqz", 100000000000 ),
+    testlib.Wallet( "5K9LmMQskQ9jP1p7dyieLDAeB6vsAj4GK8dmGNJAXS1qHDqnWhP", 100000000000 ),
+    testlib.Wallet( "5KcNen67ERBuvz2f649t9F2o1ddTjC5pVUEqcMtbxNgHqgxG2gZ", 100000000000 )
+]
+
+consensus = "17ac43c1d8549c3181b200f1bf97eb7d"
+
+def scenario( wallets, **kw ):
+
+    testlib.blockstack_namespace_preorder( "miner", wallets[1].addr, wallets[0].privkey )
+    testlib.next_block( **kw )
+
+    testlib.blockstack_namespace_reveal( "miner", wallets[1].addr, 52595, 250, 4, [6,5,4,3,2,1,0,0,0,0,0,0,0,0,0,0], 10, 10, wallets[0].privkey )
+    testlib.next_block( **kw )
+
+    testlib.blockstack_namespace_ready( "miner", wallets[1].privkey )
+    testlib.next_block( **kw )
+
+    for i in xrange(0, 25):
+        name = "foo_{}.miner".format(i)
+        testlib.blockstack_name_preorder( name, wallets[2].privkey, wallets[3].addr )
+
+    testlib.next_block( **kw )
+
+    for i in xrange(0, 25):
+        name = "foo_{}.miner".format(i)
+        testlib.blockstack_name_register( name, wallets[2].privkey, wallets[3].addr )
+        
+    testlib.next_block( **kw )
+
+    testlib.blockstack_name_preorder( "bar.miner", wallets[4].privkey, wallets[5].addr )
+    testlib.blockstack_name_register( "bar.miner", wallets[4].privkey, wallets[5].addr )
+    testlib.next_block( **kw )
+
+    # block_threshold = 500
+    block_threshold = 10
+    for i in xrange(0, block_threshold + 1):
+        testlib.next_block( **kw )
+
+def check( state_engine ):
+
+    migration_data_file_path = os.path.join( state_engine.working_dir, 'v2_migration_data.tar.bz2')
+    if not os.path.exists(migration_data_file_path):
+        print 'v2_migration_data file not found'
+        return False
+
+    # not revealed, but ready 
+    ns = state_engine.get_namespace_reveal( "miner" )
+    if ns is not None:
+        print "namespace not ready"
+        return False 
+
+    ns = state_engine.get_namespace( "miner" )
+    if ns is None:
+        print "no namespace"
+        return False 
+
+    if ns['namespace_id'] != 'miner':
+        print "wrong namespace"
+        return False 
+
+    # not preordered
+    preorder = state_engine.get_name_preorder( "foo_1.miner", virtualchain.make_payment_script(wallets[2].addr), wallets[3].addr )
+    if preorder is not None:
+        print "still have preorder"
+        return False
+    
+    # registered 
+    name_rec = state_engine.get_name( "foo_1.miner" )
+    if name_rec is None:
+        print "name does not exist"
+        return False 
+
```

Comment:
Do you want to maybe (1) advance a few more blocks in the `scenario()` method (via `testlib.next_block(**kw)`), and (2) inspect the snapshot tarball here and see that the snapshots database it contains has processed only up to the activation threshold block height?

Reply:
Done -- added this to the integration test

Reply:
Cool, saw that.  Will `bar.miner` be in the fast-sync snapshot, or is it excluded because the threshold was met after `foo_25.miner`?

Reply:
`bar.miner` is in the snapshot because it's after the name threshold is reached, but before the additional 500 blocks (set to 10 blocks in testing env) have passed to trigger the export, per the [spec](https://paper.dropbox.com/doc/Stacks-v1-v2-Upgrade-Implementation--A~DogOYhIkjj_YmSD9zEz2eMAg-bwo5CumbrcSGFvm3NBuzc). 

I just added an additional name registration after this block, and the corresponding test to ensure it is excluded from the snapshot 👍

Reply:
Great, thanks!